src/ZeroC.Slice/Internal/Activator.cs

```diff
@@ -100,28 +89,6 @@ internal Activator Get(Assembly assembly)
             // We don't cache an assembly with no Slice attribute, and don't load/process its referenced assemblies.
             return Activator.Empty;
         }
-
-        static ActivateObject CreateActivateObject(Type type)
-        {
-            bool isException = type.IsSubclassOf(typeof(SliceException));
-
-            Type[] types = isException ? [typeof(string), typeof(Exception)] : [];
-
-            ConstructorInfo constructor = type.GetConstructor(
-                BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic,
-                null,
-                types,
-                null) ?? throw new InvalidOperationException($"Cannot find Slice decoding constructor for '{type}'.");
-
-            ParameterExpression messageParam = Expression.Parameter(typeof(string), "message");
-            ParameterExpression innerExceptionParam = Expression.Parameter(typeof(Exception), "innerException");
-
-            Expression expression = isException ?
-                Expression.New(constructor, messageParam, innerExceptionParam) :
-                Expression.New(constructor);
-
-            return Expression.Lambda<ActivateObject>(expression, messageParam, innerExceptionParam).Compile();
```

Comment:
Using `Expression.Lambda`  is much faster than  `Activator.CreateInstance`

Reply:
It's surprising, and more importantly, does it matter?

Reply:
> It's surprising,

One uses compiled code and the other uses reflection.

>  does it matter?

It depends, this would be noticeable for applications decoding a larger number of classes. So probably not very common.

It is a tradeoff of simpler code to this extra bit of performance for class decoding.

Reply:
Here is benchmark:
https://andrewlock.net/benchmarking-4-reflection-methods-for-calling-a-constructor-in-dotnet/

If we keep the "compiled expression" approach, we also need to keep LazyObject.

Note that ActivatorCreateInstance is not as slow as actual Reflection.

Reply:
I'm fine with using the simpler CreateInstance, I just wrote the comment so that we were aware of the performance differences.